sql/descriptor.go

```diff
@@ -62,12 +62,15 @@ type descriptorProto interface {
 }
 
 // checkPrivilege verifies that p.user has `privilege` on `descriptor`.
-func (p *planner) checkPrivilege(descriptor descriptorProto, privilege privilege.Kind) error {
-	if descriptor.GetPrivileges().CheckPrivilege(p.user, privilege) {
+func (p *planner) checkPrivilege(descriptor descriptorProto, priv privilege.Kind) error {
+	if descriptor.GetPrivileges().CheckPrivilege(p.user, priv) {
 		return nil
 	}
+	if priv != privilege.SELECT && isSystemConfigID(descriptor.GetID()) {
+		return fmt.Errorf("system database is read-only")
```

Comment:
Yeah, I don't want to speak for @jseldess too much, but my impression was that the suggestion in #4676 was just to change the error to stop you from thinking that you'd be able to get around the privilege error by going and granting it.